plugins/guests/debian/cap/configure_networks.rb

```diff
@@ -61,12 +61,21 @@ def self.configure_netplan(machine, interfaces, comm, networks)
           # By default, netplan expects the renderer to be systemd-networkd,
           # but if any device is managed by NetworkManager, then we use that renderer
           # ref: https://netplan.io/reference
-          renderer = NETPLAN_DEFAULT_RENDERER
-          ethernets.keys.each do |k|
-            if nm_controlled?(comm, k)
-              renderer = "NetworkManager"
-              break
+          if networkd?(comm)
```

Comment:
netplan supports `networkd` and `NetworkManager` as renderers (https://netplan.io/), by default it uses `networkd`. If `networkd` is not installed, Vagrant should probably fall back to `NetworkManager` if it is available. If neither of these are available, then netplan will not work (Vagrant should throw an error). 
Additionally, a device can be set to be managed by `NetworkManager`, in which case netplan should use the `NetworkManager` renderer.